internal/server/proto/server.proto

```diff
@@ -2096,6 +2136,16 @@ message Deployment {
   }
 }
 
+// A deployment with additional related messages pre-fetched.
+message DeploymentExpanded {
```

Comment:
Should we include artifact too like how pre-load does?

Reply:
@jgwhite Anything else we should include here for the UI?

Reply:
I think this is the right place to start, for sure.

Am I right in thinking nothing is returning these types yet?

Reply:
Right you are. That isn't implemented yet - this wip was intended just to demonstrate internal wiring of declared resources.

Reply:
I just thought of one other preload-y thing we definitely will want right away: `deploy_url`

Reply:
I like the idea of eventually deprecating all the fields in the Preload sections and moving them out to the Expanded apis. We can address that in another PR.

Reply:
I'll leave this in here for now, but I think we will pretty shortly replace DeploymentExpanded with some different methods aimed directly at serving the UI.